feature/signupflow/onboarding/src/main/java/com/epmedu/animeal/signup/onboarding/presentation/OnboardingViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.epmedu.animeal.signup.onboarding.presentation
+
+import android.app.Activity
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.amplifyframework.core.Amplify
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.DefaultEventDelegate
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.EventDelegate
+import com.epmedu.animeal.profile.data.mapper.AuthUserAttributesToProfileMapper
+import com.epmedu.animeal.profile.data.model.SocialLoginListener
+import com.epmedu.animeal.profile.domain.SaveProfileUseCase
+import com.epmedu.animeal.profile.domain.authentication.FacebookSignInUseCase
+import com.epmedu.animeal.signup.onboarding.presentation.OnboardingViewModel.Event
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+internal class OnboardingViewModel @Inject constructor(
+    private val facebookSignInUseCase: FacebookSignInUseCase,
+    private val saveProfileUseCase: SaveProfileUseCase,
+    private val authUserAttributesToProfileMapper: AuthUserAttributesToProfileMapper
+) : ViewModel(),
+    EventDelegate<Event> by DefaultEventDelegate() {
+
+    fun authenticateByFacebook(activity: Activity) {
```

Comment:
Would appreciate some advice on how to cleanly pass activity to the FacebookSignInUseCase and AuthenticationRepositoryImpl

Reply:
Seems to be impossible to make. If we are creating the repository outside ViewModelComponent, ViewModel cannot inject it. If we are creating the repository in ViewModelComponent, it's impossible to inject activity or activity context into it.
In case it is truly impossible, my suggestion is to create separate Activity which will do socialSignIn in itself and then just return us to the previous activity. Or we can just call socialSignIn outside of viewModel, which is also very unoptimal

Reply:
Created FacebookSignInActivity which handles signInWithSocialWebUI